test/tbb/test_flow_graph_priorities.cpp

```diff
@@ -332,7 +332,7 @@ struct DeciderBody {
 
 struct AsyncSubmissionBody {
     AsyncActivity* my_activity;
-    void operator()(data_type input, async_node_type::gateway_type& gateway) {
+    void operator()(data_type input, async_node_type::gateway_type& gateway) noexcept {
```

Comment:
In last realization async_node will no execute this task immediately. Follow queuing policy if body is noexcept.

Reply:
Okay, we fount out why lightweight execution is actually needed. Let's write descriptive comment in this regard:
```suggestion
    // It is important that async_node in the test executes without spawning a TBB task, because
    // it passes the work to asynchronous thread, which unlocks the barrier that is waited
    // by every execution thread (asynchronous thread and any TBB worker or main thread).
    // This is why async_node's body marked noexcept.
    void operator()(data_type input, async_node_type::gateway_type& gateway) noexcept {
```